src/backend/opencl/compile_module.cpp

```diff
@@ -119,6 +125,21 @@ Program buildProgram(const vector<string> &kernelSources,
 
 }  // namespace opencl
 
+string getKernelCacheFilename(const int device, const string &key) {
+    auto &dev = opencl::getDevice(device);
+
+    unsigned vendorId = dev.getInfo<CL_DEVICE_VENDOR_ID>();
+    auto devName      = dev.getInfo<CL_DEVICE_NAME>();
+    string infix      = to_string(vendorId) + "_" + devName;
+
+    transform(infix.begin(), infix.end(), infix.begin(),
+              [](unsigned char c) { return std::toupper(c); });
+    std::replace(infix.begin(), infix.end(), ' ', '_');
+
+    return "KER" + key + "_CL_" + infix + "_AF_" +
+           to_string(AF_API_VERSION_CURRENT) + ".clbin";
```

Comment:
May be use git hash or something else  instead of version? It may get tricky debugging issues when using a development branch.

Reply:
That shouldn't be a problem because we hash contents of source and associated compile options for that kernel - this hash is included via `key`.